src/Server.js

```diff
@@ -54,7 +54,22 @@ function middleware(bundler) {
   };
 }
 
-function serve(bundler, port) {
+function getFreePort(port) {
+  return new Promise((resolve, reject) => {
+    let server = http.createServer().listen(port);
+    server.once('error', err => {
+      resolve(0);
```

Comment:
I think because you're resolving the `Promise` on error, it will return early and `port` will now be 0. So essentially, `getFreePort` will always return 0 for `port` if the provided port is in use.

Correct me if I'm wrong tho 😅

Reply:
You're correct this is the point though because node.js's net library sees port 0 as pick a free port which resolves into a random free port

Reply:
I'll change some stuff around to make it more clear in the code

Reply:
TIL a bit more about `net`. Thanks for that @DeMoorJasper! 😁

Reply:
@yeskunall u can read all about it here: https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback =D